presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -15,259 +15,196 @@
-            JsonCodec<TaskUpdateRequest> taskUpdateRequestCodec,
-            JsonCodec<PlanFragment> planFragmentCodec,
-            JsonCodec<TaskInfo> taskInfoCodec)
+            ScheduledExecutorService updateScheduledExecutor)
     {
         requireNonNull(session, "session is null");
         requireNonNull(taskId, "taskId is null");
         requireNonNull(planFragment, "planFragment is null");
         requireNonNull(sources, "sources is null");
         requireNonNull(httpClient, "httpClient is null");
         requireNonNull(tableWriteInfo, "tableWriteInfo is null");
-        requireNonNull(stats, "stats is null");
         requireNonNull(executor, "executor is null");
         requireNonNull(updateScheduledExecutor, "updateScheduledExecutor is null");
-        requireNonNull(taskUpdateRequestCodec, "taskUpdateRequestCodec is null");
-        requireNonNull(planFragmentCodec, "planFragmentCodec is null");
-        requireNonNull(taskInfoCodec, "taskInfoCodec is null");
 
         try (SetThreadName ignored = new SetThreadName("NativeExecutionTask-%s", taskId)) {
             this.taskId = taskId;
             this.session = session;
             this.planFragment = planFragment;
-            this.httpClient = httpClient;
             this.tableWriteInfo = tableWriteInfo;
-            this.stats = stats;
             this.sources = sources;
             this.executor = executor;
             this.updateScheduledExecutor = updateScheduledExecutor;
             this.outputBuffers = createInitialEmptyOutputBuffers(PARTITIONED);
-            List<BufferInfo> bufferStates = outputBuffers.getBuffers()
-                    .keySet().stream()
-                    .map(outputId -> new BufferInfo(outputId, false, 0, 0, PageBufferInfo.empty()))
-                    .collect(toImmutableList());
-            this.taskInfo.set(createInitialTask(taskId, URI.create(LOCAL_URI + taskId), bufferStates, new TaskStats(DateTime.now(), null), ""));
-            this.taskUpdateRequestCodec = taskUpdateRequestCodec;
-            this.planFragmentCodec = planFragmentCodec;
-            this.taskInfoCodec = taskInfoCodec;
+            this.workerClient = new PrestoSparkHttpWorkerClient(httpClient, taskId, location);
+            this.taskInfoFetcher = new HttpNativeExecutionTaskInfoFetcher(
+                    this.updateScheduledExecutor,
+                    this.workerClient,
+                    this.executor,
+                    this.taskId);
+            this.taskResultFetcher = new HttpNativeExecutionTaskResultFetcher(
+                    this.updateScheduledExecutor,
+                    this.workerClient,
+                    this.taskId,
+                    Optional.empty());
         }
     }
 
+    /**
+     * Gets the most updated TaskInfo of the task of the native task.
+     *
+     * @return most updated TaskInfo, null if TaskInfoFetcher has not yet retrieved the very first TaskInfo.
```

Comment:
It's not the `TaskInfoFetcher` right? It's something else

Reply:
Comments not addressed? https://github.com/prestodb/presto/pull/18458/files#r989704240

Reply:
Cool, I'm changing the return type to be Optional<TaskInfo>. Also changed the comments to have @link syntax